tests/posix/common/src/pthread.c

```diff
@@ -465,6 +465,40 @@ ZTEST(pthread, test_pthread_cleanup)
 	zassert_ok(pthread_join(th, NULL));
 }
 
+static bool testcancel_failed;
```

Comment:
we should initialize it to `true` here and only set it to `false` when we expect the next call (L486) to terminate the thread, so that if any of the functions (`pthread_cancel`, `pthread_setcancelstate`) below mistakenly terminates the thread, the test will fail.

Reply:
> we should initialize it to `true` here and only set it to `false` when we expect the next call (L486) to terminate the thread, so that if any of the functions (`pthread_cancel`, `pthread_setcancelstate`) below mistakenly terminates the thread, the test will fail.

I wouldn't initialize it to true (static values should typically just be zero initialized with BSS). If anything, maybe earlier in the _fn`

Reply:
I wouldn't initialize this to true.

Maybe add another static bool to very the first call to testcancel.